static/app/components/profiling/functionsMiniGrid.tsx

```diff
@@ -0,0 +1,121 @@
+import {CSSProperties, Fragment} from 'react';
+import {Link} from 'react-router';
+import styled from '@emotion/styled';
+
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import PerformanceDuration from 'sentry/components/performanceDuration';
+import {Flex} from 'sentry/components/profiling/flex';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization, Project} from 'sentry/types';
+import {SuspectFunction} from 'sentry/types/profiling/core';
+import {generateProfileFlamechartRouteWithQuery} from 'sentry/utils/profiling/routes';
+
+interface FunctionsMiniGridProps {
+  functions: SuspectFunction[] | null;
+  organization: Organization;
+  project: Project;
+}
+
+export function FunctionsMiniGrid(props: FunctionsMiniGridProps) {
+  const {organization, project, functions} = props;
+
+  const linkToFlamechartRoute = (
```

Comment:
I would move this outside of the react component and rename to linkToFlamechartWithHighlightFrame or something. You can also add it to the route utils and make frameName and framePackage a required parameter which I think is what you want here

Reply:
@eliashussary I see you resolved this but kept code the same, did you mean to move it :D?

Reply:
I did move it... it just doesn't change much of how it's used at the call site. We now have an extra 'generate' function in the routes util that explicitly accepts frameName and framePackage. That's called from here. 

If it's really bothering you I will remove this 'linkTo' fn and call the 'generate' fn directly. 

I think our routing needs some more work as a whole so I didn't want to get too caught up in it